easybuild/easyblocks/generic/configuremake.py

```diff
@@ -67,6 +67,101 @@
 DEFAULT_INSTALL_CMD = 'make install'
 
 
+def check_config_guess(config_guess, log=None):
+    """Check timestamp & SHA256 checksum of config.guess script.
+
+    :param configu_gess: Path to config.guess script to check
+    :param log: Instance of a logger to use or None to run in quiet mode
+    :return: Whether the script is valid (matches the version and checksum)
+    """
+    # config.guess includes a "timestamp='...'" indicating the version
+    config_guess_version = None
+    version_regex = re.compile("^timestamp='(.*)'", re.M)
+    res = version_regex.search(read_file(config_guess))
+    if res:
+        config_guess_version = res.group(1)
+
+    config_guess_checksum = compute_checksum(config_guess, checksum_type=CHECKSUM_TYPE_SHA256)
+    try:
+        config_guess_timestamp = datetime.fromtimestamp(os.stat(config_guess).st_mtime).isoformat()
+    except OSError as err:
+        if log:
+            log.warning("Failed to determine timestamp of %s: %s", config_guess, err)
+        config_guess_timestamp = None
+
+    # log version, timestamp & SHA256 checksum of config.guess
+    if log:
+        log.info("config.guess version: %s (last updated: %s, SHA256 checksum: %s)",
+                 config_guess_version, config_guess_timestamp, config_guess_checksum)
+
+    result = True
+
+    if config_guess_version != CONFIG_GUESS_VERSION:
+        result = False
+        tup = (config_guess, config_guess_version, CONFIG_GUESS_VERSION)
+        if log:
+            log.warning("config.guess version at %s does not match expected version: %s vs %s" % tup)
```

Comment:
stick to using `print_warning` here (which is a verbose warning, as opposed to just a warning in the log file)?

Reply:
Where is the warning function (from log) defined? Can it also print to stderr? Because print_warning (inconsistently) does not accept a log param to also print to log.

Reply:
See https://github.com/easybuilders/easybuild-framework/pull/3195